### PR TITLE
ARROW-6026: [Doc] Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Review by kou — APPROVED** [review]
+1

How about adding a link to https://arrow.apache.org/docs/developers/contributing.html ?

**Comment by pitrou** [comment]
Good idea, I'll do that.

**Comment by pitrou** [comment]
Updated.

File: docs/source/developers/contributing.rst [hunk 1]
```diff
@@ -52,23 +52,38 @@ In addition, the project's `Confluence site <https://cwiki.apache.org/confluence
...
+  field. For feature requests and other proposals, "Affects version" isn't
+  appropriate.
 
 Project maintainers may later tweak formatting and labels to help improve their
 visibility. They may add a "Fix version" to indicate that they're considering
 it for inclusion in the next release, though adding that tag is not a
 commitment that it will be done in the next release.
 
+Advanced use
+^^^^^^^^^^^^
```
Comment (kou) [inline docs/source/developers/contributing.rst]: Should we use `---` instead of `^^^` because the following "GitHub issues" section uses `---`?

Comment (pitrou) [reply docs/source/developers/contributing.rst]: Right. Will fix.

**Comment by kou** [comment]
I'll merge this.

**Pull request closed without merging**